language/diem-vm/src/diem_transaction_executor.rs

```diff
@@ -57,12 +57,12 @@ impl DiemVM {
 
     /// Generates a transaction output for a transaction that encountered errors during the
     /// execution process. This is public for now only for tests.
-    pub fn failed_transaction_cleanup(
+    pub fn failed_transaction_cleanup<R: RemoteCache>(
         &self,
         error_code: VMStatus,
         gas_status: &mut GasStatus,
         txn_data: &TransactionMetadata,
-        remote_cache: &StateViewCache<'_>,
+        remote_cache: &R,
```

Comment:
It might be a good time to rename a few things:
- Parameter name "remote_cache" to something like "storage"
- Trait name "RemoteCache" to something like "MoveStorage"
  - Type parameter "R"